core/templates/dev/head/pages/topic_editor/TopicEditorStateService.js

```diff
@@ -209,11 +212,43 @@ oppia.factory('TopicEditorStateService', [
        * _subtopicPages list.
        */
       setSubtopicPage: function(subtopicPage) {
-        if (_getSubtopicPageIndex(subtopicPage.getId()) !== -1) {
+        if (_getSubtopicPageIndex(subtopicPage.getId()) !== null) {
           _subtopicPages[_getSubtopicPageIndex(subtopicPage.getId())] =
             angular.copy(subtopicPage);
         } else {
           _setSubtopicPage(subtopicPage);
+          _newSubtopicPages.push(angular.copy(subtopicPage));
+        }
+      },
+
+      deleteSubtopicPage: function(topicId, subtopicId) {
+        var subtopicPageId = topicId + '-' + subtopicId.toString();
+        var index = _getSubtopicPageIndex(subtopicPageId);
+        var newIndex = _getNewSubtopicPageIndex(subtopicPageId);
+        if (index !== null) {
+          _subtopicPages.splice(index, 1);
+          // If the deleted subtopic page corresponded to a newly created
+          // subtopic, then the 'subtopicId' part of the id of all subsequent
+          // subtopic pages should be decremented to make it in sync with the
+          // their corresponding subtopic ids.
+          if (newIndex !== null) {
```

Comment:
Just to check, should we be updating the "next subtopic id" property of the topic, too?

Reply:
The next subtopic id and other related properties of the subtopic are modified in TopicUpdateService itself. Here, the related changes are done to subtopic page.